### DISCUSSION TITLE
load data from external databases with the desktop app

### DISCUSSION THREAD

**Discussion by SzymonFabian** [whitphx/stlite#761] [Q&A]
Hi, 

is it possible to load data from the internet with the desktop application? 

I have this type of error:
AttributeError: module 'http.client' does not have attribute 'HTTPSConnection'
Traceback:
File 'streamlit/runtime/scriptrunner/script_runner.py', line 542, in _run_script
File '/home/pyodide/streamlit_app.py', line 13, in <module>
    db = deta.Base("XXX")
         ^^^^^^^^^^^^^^^^^^^^^^^^^
File "/lib/python3.11/site-packages/deta/__init__.py", line 43, in Base
    return _Base(name, self.project_key, self.project_id, host)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/lib/python3.11/site-packages/deta/base.py", line 76, in __init__
    super().__init__(
File "/lib/python3.11/site-packages/deta/service.py", line 27, in __init__
    self.client = http.client.HTTPSConnection(host, timeout=timeout) if keep_alive else None

**Comment by NoorCAE** [comment]
i am working on connecting the desktop app to an external database, do you know of any other way for the app to load data? I tried putting an excel file on the streamlit_app directory while building and while it loads the data it dosent save the data as i imagine it only loads an image (kinda like docker) so i thought connecting an external database would allow to CRUD. please let me know if you loading data from external databases also allows you to write back into it
